Cargo.toml

```diff
@@ -30,37 +30,52 @@ categories = [
 
 keywords = ["frontmatter", "yaml", "toml", "json", "frontmatter-gen"]
 
+# The library file that contains the main logic for the binary.
 [lib]
 name = "frontmatter_gen"
 path = "src/lib.rs"
 
+# The main file that contains the entry point for the binary.
+[[bin]]
+name = "frontmatter_gen"
+path = "src/main.rs"
+
 # -----------------------------------------------------------------------------
 # Dependencies
 # -----------------------------------------------------------------------------
 
 [dependencies]
-
-serde = { version = "1.0", features = ["derive"] }
-serde_json = "1.0"
+anyhow = "1.0.93"
+clap = { version = "4.5.21", features = ["derive", "color", "help", "suggestions"] }
+dtt = "0.0.8"
+log = "0.4.22"
+pretty_assertions = "1.4.1"
+pulldown-cmark = "0.12.2"
+serde = { version = "1.0.215", features = ["derive"] }
+serde_json = "1.0.132"
 serde_yml = "0.0.12"
-thiserror = "2.0"
-toml = "0.8"
-tokio = { version = "1.0", features = ["full"] }
+tera = "1.20.0"
+thiserror = "2.0.3"
+tokio = { version = "1.41.1", features = ["full"] }
+toml = "0.8.19"
+url = "2.5.3"
+uuid = { version = "1.11.0", features = ["v4", "serde"] }
```

Comment:
Hi,

I see you are adding an embedded SSG, which is cool, however when the intention is to use this lib as a front-matter parser only (in my case I use comrak for markdown) it will bring in more deps and features than I need.

Have you considered gating the ssg under a feature?

```toml
[dependencies]
frontmatter_gen = {version = "*", features = ["ssg"]}
```

This way you can keep the functionality but allow the lib to still be used as a simple frontmatter parser

Reply:
@rochacbruno yeah good idea. Thanks for that